cpp/src/arrow/io/hdfs.cc

```diff
@@ -497,12 +501,15 @@ class HadoopFileSystem::HadoopFileSystemImpl {
   Status OpenReadable(const std::string& path, int32_t buffer_size,
                       const io::IOContext& io_context,
                       std::shared_ptr<HdfsReadableFile>* file) {
+    errno = 0;
     hdfsFile handle = driver_->OpenFile(fs_, path.c_str(), O_RDONLY, buffer_size, 0, 0);
 
     if (handle == nullptr) {
-      const char* msg = !Exists(path) ? "HDFS file does not exist: "
-                                      : "HDFS path exists, but opening file failed: ";
-      return Status::IOError(msg, path);
+      if (errno) {
```

Comment:
Dummy question, since the `errno` variable is never passed to a libhdfs method, how can this ever have another value than 0?

Reply:
This is the pseudo-magic C `errno` variable :-)
See https://pubs.opengroup.org/onlinepubs/9699919799/functions/errno.html, https://en.cppreference.com/w/c/error/errno

Reply:
Aha, I see. I assume that's standard enough C knowledge to not require a comment ;)

No further comments on the PR!